spec/controllers/v0/account_controls_controller_spec.rb

```diff
@@ -0,0 +1,288 @@
+
+        it 'returns an internal server error' do
+          post lock_action, params: account_controls_params
+
+          expect(response).to have_http_status(:internal_server_error)
+          expect(JSON.parse(response.body)['error']).to eq(expected_error_message)
+        end
+
+        it 'logs the lock attempt' do
+          allow(Rails.logger).to receive(:info)
+          expect(Rails.logger).to receive(:info).with("UserAccount CSP #{expected_lock_action} failed",
+                                                      expected_response_log)
+
+          post lock_action, params: account_controls_params
+        end
+      end
+    end
+
+    context 'when a request is made without a type' do
+      let(:type_param) { nil }
+      let(:expected_error_message) { 'The required parameter "type", is missing' }
+
+      it 'returns a type parameter missing error' do
+        post lock_action, params: account_controls_params
+
+        expect(response).to have_http_status(:bad_request)
+        expect(JSON.parse(response.body)['errors'].first['detail']).to eq(expected_error_message)
+      end
+    end
+
+    context 'when a request is made with a type that is not found in VALID_CSP_TYPES' do
+      let(:type_param) { 'some-csp-type' }
+      let(:expected_error_message) { "\"#{type_param}\" is not a valid value for \"type\"" }
+
+      it 'returns a type parameter missing error' do
+        post lock_action, params: account_controls_params
+
+        expect(response).to have_http_status(:bad_request)
+        expect(JSON.parse(response.body)['errors'].first['detail']).to eq(expected_error_message)
+      end
+    end
+
+    context 'when a request is made without a CSP uuid' do
+      let(:csp_uuid_param) { nil }
+      let(:expected_error_message) { 'The required parameter "csp_uuid", is missing' }
+
+      it 'returns a CSP uuid parameter missing error' do
+        post lock_action, params: account_controls_params
+
+        expect(response).to have_http_status(:bad_request)
+        expect(JSON.parse(response.body)['errors'].first['detail']).to eq(expected_error_message)
+      end
+    end
+
+    context 'when a request is made with a CSP uuid' do
+      context 'when a UserVerification matches the requested CSP uuid' do
+        it_behaves_like 'when a record is found'
+      end
+
+      context 'when a UserVerification does not match the requested CSP uuid' do
+        let(:csp_uuid_param) { 'some-csp-uuid' }
+
+        it_behaves_like 'when a record is not found'
+      end
+    end
+  end
+
+  describe 'POST csp_lock' do
+    let(:lock_action) { :csp_lock }
+    let(:locked) { false }
+    let(:type_param) { type }
+    let(:csp_uuid_param) { csp_uuid }
+    let(:account_controls_params) { { type: type_param, csp_uuid: csp_uuid_param } }
+    let(:expected_lock_status) { true }
+    let(:expected_lock_action) { lock_action == :csp_lock ? 'lock' : 'unlock' }
```

Comment:
I wouldn't put conditionals in a `let` block, instead redefine this `let` in the appropriate `context` block